src/ui/hook/useItemTypeColor.ts

```diff
@@ -0,0 +1,19 @@
+import { useCallback } from 'react'
+
+export type DataverseItem = 'dataspace' | 'dataset' | 'service'
+export type ColorVariant = 'primary-color' | 'primary-color-variant-3' | 'primary-color-variant-4'
+
+export const useItemTypeColor = (type: DataverseItem): ColorVariant => {
+  const renderItemTypeColor = useCallback((type: DataverseItem): ColorVariant => {
+    switch (type) {
+      case 'service':
+        return 'primary-color'
+      case 'dataspace':
+        return 'primary-color-variant-3'
+      case 'dataset':
+        return 'primary-color-variant-4'
+    }
+  }, [])
+
+  return renderItemTypeColor(type)
+}
```

Comment:
I don't really understand the usage of a hook here?
A hook mainly abstracts some generic logic that generates side effects like DOM events, state management, querying.. 
Here is more an utility function that handles style with a given parameter whose type is restricted to a `DataverseItem`.
IMHO, it makes more sense to extract this logic into a scss `mixin` that will be used with secondary classNames like `dataset`, `service` or `dataspace`